src/elementary/src/overload.hpp

```diff
@@ -0,0 +1,20 @@
+#ifndef NJOY_ELEMENTARY_OVERLOAD
+#define NJOY_ELEMENTARY_OVERLOAD
+
+// system includes
+
+// other includes
+
+namespace njoy {
+namespace elementary {
+
+  template < class... Types > struct overload : Types... {
+
+    using Types::operator()...;
+  };
+  template < class... Types > overload( Types... ) -> overload< Types... >;

```

Comment:
I'm a little surprised you can do this. It seems that you have a struct and a function with the same name `overload`. How does the compiler know the difference? Should the struct be named `Overload` to match our style guide?